overlord/cmdstate/cmdmgr.go

```diff
@@ -46,28 +46,38 @@ func (m *CommandManager) Ensure() error {
 var defaultExecTimeout = 5 * time.Second
 
 func doExec(t *state.Task, tomb *tomb.Tomb) error {
-	var argv []string
-	var tout time.Duration
-
 	st := t.State()
 	st.Lock()
-	err1 := t.Get("argv", &argv)
-	err2 := t.Get("timeout", &tout)
-	st.Unlock()
-	if err1 != nil {
-		return err1
+	defer st.Unlock()
+
+	var ignore bool
+	if err := t.Get("ignore", &ignore); err != nil && err != state.ErrNoState {
+		return err
+	}
+	if ignore {
+		t.Logf("task ignored")
+		return nil
+	}
+
+	var argv []string
+	var tout time.Duration
+	if err := t.Get("argv", &argv); err != nil {
+		return err
 	}
-	if err2 != nil && err2 != state.ErrNoState {
-		return err2
+
+	err := t.Get("timeout", &tout)
+	if err != nil && err != state.ErrNoState {
+		return err
 	}
-	if err2 == state.ErrNoState {
+	if err == state.ErrNoState {
 		tout = defaultExecTimeout
 	}
 
-	if buf, err := osutil.RunAndWait(argv, nil, tout, tomb); err != nil {
-		st.Lock()
+	st.Unlock()
```

Comment:
I have a slight preference for consistency in locking/unlocking within a single function, so I think we should follow the case of above where we Lock then defer an Unlock, and so here I think we should Unlock, then defer an Lock so the previous Unlock() works.

Reply:
The problem is that Errorf needs a lock. See comment below.